packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -81,17 +82,19 @@ contract TaikoL2 is EssentialContract, IXchainSync {
      * @param l1Height The latest L1 block height when this block was proposed.
      * @param l1Hash The latest L1 block hash when this block was proposed.
      * @param l1SignalRoot The latest value of the L1 "signal service storage root".
+     * @return parentHash The parent hash. Circuits need to check this hash is
+     *         the same as the block header's parentHash.
      */
     function anchor(
         uint256 l1Height,
         bytes32 l1Hash,
         bytes32 l1SignalRoot
-    ) external {
+    ) external returns (bytes32 parentHash) {
```

Comment:
@Brechtpd I think before this change, we are not verifying that the `blockhash(block.number-1)` on L2 has the same value as the `parentHash` in the block header.  The circuit needs to verify the return value of `anchor` matches the `parentHash` in the block header. Does it?